dwi_ml/data/hdf5_creation.py

```diff
@@ -0,0 +1,242 @@
+# -*- coding: utf-8 -*-
+import glob
+import logging
+from pathlib import Path
+from typing import List
+
+from dipy.io.stateful_tractogram import (set_sft_logger_level, Space)
+from dipy.io.streamline import load_tractogram
+from dipy.tracking.utils import length
+import nibabel as nib
+import numpy as np
+from scilpy.tracking.tools import resample_streamlines_step_size
+from scilpy.utils.streamlines import compress_sft
+
+from dwi_ml.data.processing.dwi.dwi import standardize_data
+
+
+def verify_subject_lists(dwi_ml_folder: Path, chosen_subjs: List[str]):
+    """
+    Raises error if some subjects in training set or validation set don't
+    exist.
+
+    Prints logging info if some subjects in dwi_ml folder were not chosen
+    in training set nor validation set.
+    """
+
+    # Find list of existing subjects from folders
+    all_subjs = [str(f.name) for f in dwi_ml_folder.iterdir() if f.is_dir()]
+    if len(all_subjs) == 0:
+        raise ValueError('No subject found in dwi_ml folder: '
+                         '{}'.format(dwi_ml_folder))
+
+    # Checking chosen_subjs
+    non_existing_subjs = [s for s in chosen_subjs if s not in all_subjs]
+    if len(non_existing_subjs) > 0:
+        raise ValueError('Following subjects were chosen either for '
+                         'training set or validation set but their folders '
+                         'were not found in dwi_ml: '
+                         '{}'.format(non_existing_subjs))
+
+    ignored_subj = [s for s in all_subjs if s not in chosen_subjs]
+    if len(ignored_subj) > 0:
+        logging.info("Careful! NOT processing subjects {} "
+                     "because they were not included in training set nor "
+                     "validation set!".format(ignored_subj))
+
+    # Note. good_chosen_subjs = [s for s in all_subjs if s in chosen_subjs]
+
+
+def _load_and_check_volume_to4d(data_file):
+    """Load nibabel data, and perform some checks:
+    - Data must be Nifti
+    - Data must be at least 3D
+    Final data will most probably be 4D (3D + features). Sending loaded data to
+    4D if it is 3D.
+    """
+    ext = data_file.suffix
+
+    if ext != '.gz' and ext != '.nii':
+        raise ValueError('All data files should be nifti (.nii or .nii.gz) '
+                         'but you provided {}. Please check again your config '
+                         'file.'.format(data_file))
+
+    img = nib.load(data_file)
+    data = img.get_fdata(dtype=np.float32)
+    affine = img.affine
+    header = img.header  # Note. affine is header.get_sform().
+    img.uncache()
+
+    if len(data.shape) < 3:
+        raise NotImplementedError('Why would a data be less than 3D? We did '
```

Comment:
That error message might be a bit too "familiar", maybe something like " Data less than 3D is not handled" would be more appropriate.